applications/GeoMechanicsApplication/custom_processes/apply_c_phi_reduction_process.hpp

```diff
@@ -0,0 +1,204 @@
+                             << part_properties[INDEX_OF_UMAT_PHI_PARAMETER] << std::endl;
+            }
+            const double phi =
+                part_properties[UMAT_PARAMETERS][part_properties[INDEX_OF_UMAT_PHI_PARAMETER] - 1];
+            if (phi < 0. || phi > 90.) {
+                KRATOS_ERROR << "Friction angle Phi out of range: " << phi << std::endl;
+            }
+            return phi;
+        }
+    }
+
+    double ComputeReducedPhi(double Phi) const
+    {
+        // Phi converted to radians and then its tangent is reduced by the reduction factor
+        const double tan_phi         = std::tan(MathUtils<>::DegreesToRadians(Phi));
+        const double reduced_tan_phi = tan_phi * mReductionFactor;
+        return std::atan(reduced_tan_phi) * 180.0 / Globals::Pi; // add to MathUtils?
+    }
+
+    double GetAndCheckC(const Element::PropertiesType& rProp)
+    {
+        // Get the initial properties from the model part. Recall that we create a separate
+        // properties object with reduced c and phi for each and every element. Those reduced
+        // properties objects are not linked to the original ones.
+        const auto& part_properties = mrModelPart.GetProperties(rProp.Id());
+
+        if (!part_properties.Has(INDEX_OF_UMAT_C_PARAMETER) ||
+            !part_properties.Has(NUMBER_OF_UMAT_PARAMETERS) || !part_properties.Has(UMAT_PARAMETERS)) {
+            KRATOS_ERROR << "Insufficient material data for C-phi reduction: " << std::endl;
+        } else {
+            if (part_properties[INDEX_OF_UMAT_C_PARAMETER] < 1 ||
+                part_properties[INDEX_OF_UMAT_C_PARAMETER] > part_properties[NUMBER_OF_UMAT_PARAMETERS]) {
+                KRATOS_ERROR << "undefined INDEX_OF_UMAT_C_PARAMETER: "
+                             << part_properties[INDEX_OF_UMAT_C_PARAMETER] << std::endl;
+            }
+            const auto c = part_properties[UMAT_PARAMETERS][part_properties[INDEX_OF_UMAT_C_PARAMETER] - 1];
+            if (c < 0.) KRATOS_ERROR << "Cohesion C out of range: " << c << std::endl;
+            return c;
+        }
+    }
+
+    double GetAndCheckYoung(const Element::PropertiesType& rProp)
+    {
+        const auto young = mrModelPart.GetProperties(rProp.Id())[UMAT_PARAMETERS][0];
+        if (young < 0.)
+            KRATOS_ERROR << "Positive value expected for Youngs modulus UMAT_PARAMETERS(1) "
+                         << young << std::endl;
+        return young;
+    }
+
+    double GetAndCheckPoisson(const Element::PropertiesType& rProp)
+    {
+        const auto nu = mrModelPart.GetProperties(rProp.Id())[UMAT_PARAMETERS][1];
+        if (nu < -1. || nu >= 0.5)
+            KRATOS_ERROR
+                << "Value between -1.0 and 0.5 expected for Poissons ratio UMAT_PARAMETERS(2) "
+                << nu << std::endl;
+        return nu;
+    }
+
+    void set_C_Phi_At_Element(Element& rElement, double ReducedPhi, double ReducedC) const
```

Comment:
This function does not have the correct format yet
```suggestion
    void SetCPhiAtElement(Element& rElement, double ReducedPhi, double ReducedC) const
```

Reply:
Thank you. Fixed.